File: framework/src/main/java/com/github/mrmitew/bodylog/di/activity/ActivityModule.java
package com.github.mrmitew.bodylog.di.activity;

import android.support.v4.app.FragmentActivity;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class ActivityModule<T extends FragmentActivity> {
    protected final T activity;

    public ActivityModule(T activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    public T provideActivity() {
        return activity;
    }
}

File: framework/src/main/java/com/github/mrmitew/bodylog/framework/di/activity/ActivityModule.kt
package com.github.mrmitew.bodylog.framework.di.activity

import android.support.v4.app.FragmentActivity

import dagger.Module
import dagger.Provides

@Module
abstract class ActivityModule<out T : FragmentActivity>(protected val activity: T) {

    @Provides
    @ActivityScope
    fun provideActivity(): T {
        return activity
    }
}
